examples/vt_aux_n/inputs/main.cpp

```diff
@@ -0,0 +1,199 @@
+	{
+		lastButtonTimestamp = isobus::SystemTiming::get_timestamp_ms();
+		simulate_button_press();
+	}
+	if (isobus::SystemTiming::time_expired_ms(lastSliderTimestamp, B2_CYCLIC_DELAY))
+	{
+		lastSliderTimestamp = isobus::SystemTiming::get_timestamp_ms();
+		simulate_slider_move();
+	}
+
+	isobus::CANNetworkManager::CANNetwork.update();
+}
+
+void raw_can_glue(isobus::HardwareInterfaceCANFrame &rawFrame, void *parentPointer)
+{
+	isobus::CANNetworkManager::CANNetwork.can_lib_process_rx_message(rawFrame, parentPointer);
+}
+
+void setup()
+{
+	isobus::CANStackLogger::set_can_stack_logger_sink(&logger);
+	CANHardwareInterface::set_number_of_can_channels(1);
+	CANHardwareInterface::assign_can_channel_frame_handler(0, &canDriver);
+
+	if ((!CANHardwareInterface::start()) || (!canDriver.get_is_valid()))
+	{
+		std::cout << "Failed to connect to the socket. The interface might be down." << std::endl;
+	}
+
+	CANHardwareInterface::add_can_lib_update_callback(update_CAN_network, nullptr);
+	CANHardwareInterface::add_raw_can_message_rx_callback(raw_can_glue, nullptr);
+
+	std::this_thread::sleep_for(std::chrono::milliseconds(250));
+
+	isobus::NAME TestDeviceNAME(0);
+
+	// Make sure you change these for your device!!!!
+	// This is an example device that is using a manufacturer code that is currently unused at time of writing
+	TestDeviceNAME.set_arbitrary_address_capable(true);
+	TestDeviceNAME.set_industry_group(1);
+	TestDeviceNAME.set_device_class(0);
+	TestDeviceNAME.set_function_code(static_cast<std::uint8_t>(isobus::NAME::Function::SteeringControl));
+	TestDeviceNAME.set_identity_number(2);
+	TestDeviceNAME.set_ecu_instance(1);
+	TestDeviceNAME.set_function_instance(0);
+	TestDeviceNAME.set_device_class_instance(0);
+	TestDeviceNAME.set_manufacturer_code(64);
+	vtNameFilters.push_back(testFilter);
+
+	testPool = isobus::IOPFileInterface::read_iop_file("vtpooldata.iop");
+
+	if (0 != testPool.size())
+	{
+		std::cout << "Loaded object pool from vtpooldata.iop" << std::endl;
+	}
+	else
+	{
+		std::cout << "Failed to load object pool from vtpooldata.iop" << std::endl;
+	}
+
+	// Generate a unique version string for this object pool (this is optional, and is entirely application specific behavior)
+	std::string objectPoolHash = isobus::IOPFileInterface::hash_object_pool_to_version(testPool);
+
+	TestInternalECU = std::make_shared<isobus::InternalControlFunction>(TestDeviceNAME, 0x1D, 0);
+	TestPartnerVT = std::make_shared<isobus ::PartneredControlFunction>(0, vtNameFilters);
+	TestVirtualTerminalClient = std::make_shared<isobus::VirtualTerminalClient>(TestPartnerVT, TestInternalECU);
+	TestVirtualTerminalClient->set_object_pool(0, isobus::VirtualTerminalClient::VTVersion::Version3, testPool.data(), testPool.size(), objectPoolHash);
+	TestVirtualTerminalClient->set_auxiliary_input_model_identification_code(MODEL_IDENTIFICATION_CODE);
+
+	/// @todo Remove this once the VT client is able to know which objects are uploaded to the VT (#65)
```

Comment:
Should be removed once we can derive which input ObjectID's are there from the uploading object pool process